alg/gdaltransformer.cpp

```diff
@@ -428,6 +428,76 @@ CPLErr CPL_STDCALL GDALSuggestedWarpOutput2(GDALDatasetH hSrcDS,
     const int nInXSize = GDALGetRasterXSize(hSrcDS);
     const int nInYSize = GDALGetRasterYSize(hSrcDS);
 
+    /* ------------------------------------------------------------- */
+    /* Special case for warping on the same (or null) CRS.           */
+    /* ------------------------------------------------------------- */
+    if (pTransformArg && pfnTransformer == GDALGenImgProjTransform)
+    {
+        double adfGeoTransform[6];
+
+        if (GDALGetGeoTransform(hSrcDS, adfGeoTransform) == CE_None &&
+            !(adfGeoTransform[0] == 1.0 && adfGeoTransform[1] == 1.0 &&
+              adfGeoTransform[2] == 0.0 && adfGeoTransform[3] == 1.0 &&
+              adfGeoTransform[4] == 0.0 && adfGeoTransform[5] == 1.0))
```

Comment:
I still think this test is a band-aid to help a test case pass, but is not correct. Either we lack additional protection to not run this new code, or the failing test should be adapted if that makes sense

Reply:
but we should definitely test that adfGeoTransform[2] and adfGeoTransform[4] are 0, ie. no rotation

Reply:
Changed to only test no rotation, all test are passing (locally), waiting for CI.